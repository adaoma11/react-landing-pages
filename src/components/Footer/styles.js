import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
