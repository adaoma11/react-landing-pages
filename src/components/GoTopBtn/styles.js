import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
