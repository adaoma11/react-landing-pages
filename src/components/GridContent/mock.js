export default {
  title: 'Grid Component Example',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptate minus repellendus fugiat aliquam neque corporis! Quidem, deserunt! Rem quod quo amet doloremque, excepturi doloribus, provident nesciunt blanditiis consequuntur facere quia!`,
  textAlign: 'justify',
  hasBg: false,
  titleUpperCase: true,
};
