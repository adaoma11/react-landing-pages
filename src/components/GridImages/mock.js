export default {
  title: 'Grid Images Example',
  description: 'Uma breve descrição sobre este album',
  gridItems: [
    {
      altText: 'Imagem 1',
      imgSrc: 'https://picsum.photos/300?random=1',
    },
    {
      altText: 'Imagem 2',
      imgSrc: 'https://picsum.photos/300?random=2',
    },
    {
      altText: 'Imagem 3',
      imgSrc: 'https://picsum.photos/300?random=3',
    },
    {
      altText: 'Imagem 4',
      imgSrc: 'https://picsum.photos/300?random=4',
    },
    {
      altText: 'Imagem 5',
      imgSrc: 'https://picsum.photos/300?random=5',
    },
    {
      altText: 'Imagem 6',
      imgSrc: 'https://picsum.photos/300?random=6',
    },
  ],
  hasBg: false,
  titleUpperCase: false,
};
