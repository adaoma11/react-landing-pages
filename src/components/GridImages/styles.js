import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
