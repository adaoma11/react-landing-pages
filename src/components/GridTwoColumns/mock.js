export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Quia iusto omnis magnam libero ea neque deserunt optio aliquam, quas esse fugit nulla, quaerat tempora repellat? Cupiditate voluptatum sint suscipit facilis.`,
  textAlign: 'justify',
  imgSrc: 'vite.svg',
  hasBg: false,
  titleUpperCase: true,
};
