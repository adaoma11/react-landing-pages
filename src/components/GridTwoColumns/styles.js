import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 30px;

    @media ${theme.medias.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`;

export const Image = styled.div`
  ${({ $imgSrc }) => css`
    display: block;
    min-height: 500px;
    background-image: url(${$imgSrc});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 60%;
  `}
`;
