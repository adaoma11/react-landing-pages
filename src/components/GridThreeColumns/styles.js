import styled from 'styled-components';
import { Title as GridItemTitle } from '../Heading/styles';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  counter-reset: grid-counter;
  width: 100%;
  gap: 50px;
`;

export const GridItem = styled.div`
  margin-top: 80px;

  ${GridItemTitle} {
    position: relative;
    left: 4rem;
    margin-bottom: 30px;
  }

  ${GridItemTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 5rem;
    top: -2rem;
    left: -4rem;
    transform: rotate(7deg);
  }
`;
