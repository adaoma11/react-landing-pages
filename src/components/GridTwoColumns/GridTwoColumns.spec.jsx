import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { GridTwoColumns } from '.';
import { theme } from '../../styles/theme';
import mock from './mock';

describe('<GridTwoColumns />', () => {
  it('should render the component with all its elements', () => {
    renderTheme(<GridTwoColumns {...mock} />);
    expect(
      screen.getByRole('heading', { name: mock.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('paragraph')).toBeInTheDocument();
    expect(screen.getByRole('img', { name: mock.title })).toHaveStyle(
      `background-image: url(${mock.imgSrc})`,
    );
  });

  it('should have dark title and text when section has no background color', () => {
    renderTheme(<GridTwoColumns {...mock} />);
    expect(screen.getByRole('heading', { name: mock.title })).toHaveStyle(
      `color: ${theme.colors.primary}`,
    );
    expect(screen.getByRole('paragraph')).toHaveStyle(
      `color: ${theme.colors.primary}`,
    );
  });

  it('should have white title and text when section has background color', () => {
    renderTheme(<GridTwoColumns {...mock} hasBg />);
    expect(screen.getByRole('heading', { name: mock.title })).toHaveStyle(
      `color: ${theme.colors.white}`,
    );
    expect(screen.getByRole('paragraph')).toHaveStyle(
      `color: ${theme.colors.white}`,
    );
  });

  it('should have grid template with 1fr 1fr on normal screens', () => {
    renderTheme(<GridTwoColumns {...mock} />);
    expect(screen.getByTestId('container')).toHaveStyle(
      'grid-template-columns:1fr 1fr',
    );
  });

  it('should have grid template with 1fr on small screens', () => {
    renderTheme(<GridTwoColumns {...mock} />);
    expect(screen.getByTestId('container')).toHaveStyleRule(
      'grid-template-columns',
      '1fr',
      {
        media: '(max-width:  991px)',
      },
    );
  });

  it('should match snapshot', () => {
    //
  });
});
