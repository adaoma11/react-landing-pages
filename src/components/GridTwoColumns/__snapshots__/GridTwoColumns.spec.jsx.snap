// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should match snapshot 1`] = `
.c2 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  align-items: center;
  gap: 30px;
}

.c3 {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.c7 {
  display: block;
  min-height: 500px;
  background-image: url(vite.svg);
  background-repeat: no-repeat;
  background-position: center;
  background-size: 60%;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-height: 100vh;
  padding: 50px 0;
  background-color: #FFFFFF;
}

.c1 {
  max-width: 1200px;
  width: 100%;
  padding: 30px;
}

.c4 {
  margin-bottom: 30px;
}

.c5 {
  color: #0A1128;
  font-size: 3rem;
  margin-bottom: 1rem;
  text-transform: uppercase;
}

.c6 {
  font-family: 'Poppins',sans-serif;
  font-size: 1.5rem;
  color: #0A1128;
  text-align: justify;
  font-weight: normal;
}

@media (max-width: 991px) {
  .c2 {
    grid-template-columns: 1fr;
  }
}

@media (max-width: 768px) {
  .c5 {
    font-size: 2rem;
  }
}

<div>
  <section
    class="c0"
  >
    <article
      class="c1"
    >
      <div
        class="c2"
        data-testid="container"
      >
        <div
          class="c3"
        >
          <header
            class="c4"
          >
            <h2
              class="c5"
            >
              Grid Two Columns
            </h2>
          </header>
          <p
            class="c6"
          >
            Lorem ipsum dolor, sit amet consectetur adipisicing elit. Quia iusto omnis magnam libero ea neque deserunt optio aliquam, quas esse fugit nulla, quaerat tempora repellat? Cupiditate voluptatum sint suscipit facilis.
          </p>
        </div>
        <div
          aria-label="Grid Two Columns"
          class="c7"
          role="img"
        />
      </div>
    </article>
  </section>
</div>
`;
